src/query/src/datafusion.rs

```diff
@@ -102,6 +103,10 @@ impl QueryEngine for DatafusionQueryEngine {
         }
     }
 
+    fn describe(&self, stmt: QueryStatement, query_ctx: QueryContextRef) -> Result<Schema> {
+        self.statement_to_plan(stmt, query_ctx)?.schema()
```

Comment:
Schema at this stage might not be accurate. Follow-up operations like logical/physical optimization may change the type of columns. To get the output schema without actually executing the query, you can use `SendableRecordBatchStream::schema()` method, returned by `QueryExecutor::execute_stream()`. The execution phase happens when you collect that async stream, so dropping it after retrieving the schema won't trigger execution.

Reply:
Unfortunately postgres issues describe command on the prepared statement when there is no actaul parameter values provided. In datafusion, it's not allowed to generate physical plan with placeholder: `Physical plan does not support logical expression $1`. 

It seems I can move one-step forward to use schema from optimised logical plan. Is there a case that schema changes in physical plan?

Reply:
The only purpose we acquire the schema here is to check and fulfill the placeholders in prepare stmt, isn't it? Then we just need the schema to "start with", doesn't need to care the optimization's rewrites.

Reply:
We don't actually use this schema for now. It's Postgres clients request this schema for the resultset and they won't fetch it during actual query.

Reply:
I go through the existing physical plan optimizer and looks like the concrete type won't be changed during this phase. So maybe schema from the optimized logical plan is enough for us.

Reply:
Updated to use optimised plan for schema